File: resource-registry/src/main/java/nl/haploid/octowight/registry/repository/ResourceRootDmoFactory.java
package nl.haploid.octowight.registry.repository;

import nl.haploid.octowight.registry.data.ResourceRoot;
import org.springframework.stereotype.Component;

@Component
public class ResourceRootDmoFactory {

	public ResourceRootDmo fromResourceRoot(final ResourceRoot resourceRoot) {
		final ResourceRootDmo resourceRootDmo = new ResourceRootDmo();
		resourceRootDmo.setAtomId(resourceRoot.getAtomId());
		resourceRootDmo.setAtomLocus(resourceRoot.getAtomLocus());
		resourceRootDmo.setAtomType(resourceRoot.getAtomType());
		resourceRootDmo.setResourceId(resourceRoot.getResourceId());
		resourceRootDmo.setResourceType(resourceRoot.getResourceType());
		return resourceRootDmo;
	}
}

File: resource-registry/src/main/scala/nl/haploid/octowight/registry/repository/ResourceRootDmoFactory.scala
package nl.haploid.octowight.registry.repository

import nl.haploid.octowight.registry.data.ResourceRoot
import org.springframework.stereotype.Component

@Component
class ResourceRootDmoFactory {

  def fromResourceRoot(resourceRoot: ResourceRoot) = {
    val resourceRootDmo = new ResourceRootDmo
    resourceRootDmo.setAtomId(resourceRoot.getAtomId)
    resourceRootDmo.setAtomOrigin(resourceRoot.getAtomOrigin)
    resourceRootDmo.setAtomType(resourceRoot.getAtomType)
    resourceRootDmo.setResourceId(resourceRoot.getResourceId)
    resourceRootDmo.setResourceType(resourceRoot.getResourceType)
    resourceRootDmo
  }
}
